src/routes/edit-invoice.tsx

```diff
@@ -57,45 +69,31 @@ export default function ViewInvoice() {
     setStatus('loading');
 
     try {
-      await InvoicesService.updateInvoice(id, data);
+      await updateInvoiceMutation.mutateAsync({ id, data });
```

Comment:
Not sure if I get the idea behind status here :D You already have status form react-query, should be inside (updateInvoiceMutation)